compiler/src/yul/mappers/expressions.rs

```diff
@@ -344,79 +384,120 @@ mod tests {
+            ExpressionAttributes::new(
+                Type::Map(Map {
+                    key: foo_key,
+                    value: Box::new(foo_value.clone()),
                 }),
-                location: Location::Storage { index: 0 },
-            },
+                Location::Storage { nonce: Some(0) },
+            ),
         );
 
         harness.add_expression(
             "bar_array",
-            ExpressionAttributes {
-                typ: Type::Array(Array {
+            ExpressionAttributes::new(
+                Type::Array(Array {
                     dimension: 100,
-                    inner: Base::Byte,
+                    inner: Base::Address,
                 }),
-                location: Location::Memory,
-            },
+                Location::Memory,
+            ),
         );
 
+        harness.add_expression(
+            "index",
+            ExpressionAttributes::new(Type::Base(Base::U256), Location::Value),
+        );
+
+        let mut attributes = ExpressionAttributes::new(bar_value, Location::Memory);
+        attributes.move_location = Some(Location::Value);
+        harness.add_expression("bar_array[index]", attributes);
+
+        let mut attributes =
+            ExpressionAttributes::new(foo_value, Location::Storage { nonce: None });
+        attributes.move_location = Some(Location::Memory);
+        harness.add_expression("self.foo_map[bar_array[index]]", attributes);
+
         let result = map(&harness.context, &harness.src);
 
         assert_eq!(
             result,
-            "scopy(dualkeccak256(0, mloadn(add(bar_array, mul(index, 1)), 1)), 160)"
+            "scopy(dualkeccak256(0, mloadn(add(bar_array, mul(index, 20)), 20)), 160)"
         );
     }
 
     #[test]
     fn msg_sender() {
-        let result = map(&Context::new(), "msg.sender");
+        let mut harness = ContextHarness::new("msg.sender");
+        harness.add_expression(
+            "msg.sender",
+            ExpressionAttributes::new(Type::Base(Base::Address), Location::Value),
+        );
+
+        let result = map(&harness.context, "msg.sender");
 
         assert_eq!(result, "caller()");
     }
 
     #[rstest(
         expression,
         expected_yul,
-        case("1 + 2 ", "add(1, 2)"),
+        case("1 + 2", "add(1, 2)"),
         case("1 - 2", "sub(1, 2)"),
         case("1 * 2", "mul(1, 2)"),
         case("1 / 2", "div(1, 2)"),
         case("1 ** 2", "exp(1, 2)"),
-        case("5 % 2", "mod(5, 2)"),
+        case("1 % 2", "mod(1, 2)"),
         case("1 & 2", "and(1, 2)"),
         case("1 | 2", "or(1, 2)"),
         case("1 ^ 2", "xor(1, 2)"),
         case("1 << 2", "shl(2, 1)"),
         case("1 >> 2", "shr(2, 1)")
     )]
     fn arithmetic_expression(expression: &str, expected_yul: &str) {
-        let result = map(&Context::new(), expression);
+        let mut harness = ContextHarness::new(expression);
+        harness.add_expressions(
+            vec!["1", "2", expression],
```

Comment:
It's not obvious to me what this change is about...

Reply:
Previously, the mapper was not querying all expression attributes, so some unit tests ran fine without having attributes given to certain expressions. The `expr` mapper method now checks every expression for a `move_location`, so these unit tests need a context with all expression attributes populated.

This is using `ContextHarness` to create a mock context that has expression attributes on "1", "2", and the whole `expression`.